File: LeetCode Problems/2201 to 2300/2244. Minimum Rounds to Complete All Tasks/2244. Minimum Rounds to Complete All Tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
         map<int, int> mp;
        for(int i = 0; i < tasks.size(); i++) {
            mp[tasks[i]]++;
        }
        int cnt = 0;
        for(auto it : mp) {
            if(it.second <= 1)
                return -1;
            cnt += it.second % 3 == 0 ? it.second / 3 : (it.second / 3) + 1;
        }
        return cnt;
    }
};

File: LeetCode Problems/2201 to 2300/2244. Minimum Rounds to Complete All Tasks/2244. Minimum Rounds to Complete All Tasks.java
class Solution {
    public int minimumRounds(int[] tasks) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < tasks.length; i++) {
            if (mp.get(tasks[i]) == null)
                mp.put(tasks[i], 1);
            else
                mp.put(tasks[i], mp.get(tasks[i]) + 1);
        }
        int cnt = 0;
        for (Map.Entry<Integer, Integer> item : mp.entrySet()) {
            int value = item.getValue();
            if (value <= 1) return -1;
            cnt += value % 3 == 0 ? value / 3 : (value / 3) + 1;
        }
        return cnt;
    }
}
